build.sbt

```
@@ -1,17 +1,31 @@
 lazy val scala212 = "2.12.18"
 lazy val scala213 = "2.13.7"
+lazy val scala213dot11 = "2.13.14"
+lazy val scala3 = "3.3.4"
 
 lazy val commonSettings = List(
   scalaVersion := scala212,
-  scalacOptions ++= Seq(
-    "-encoding",
-    "utf8",
-    "-deprecation",
-    "-unchecked",
-    "-Xlint",
-    "-Xfatal-warnings"
-  ),
-  Compile / console / scalacOptions --= Seq("-deprecation", "-Xfatal-warnings", "-Xlint")
+  scalacOptions ++= {
+    val x = CrossVersion.partialVersion(scalaVersion.value) flatMap {
+      case (2, _) => Some(Seq("-Xlint"))
+      case _ => None
+    }
+    Seq(
+      "-encoding",
+      "utf8",
+      "-deprecation",
+      "-unchecked",
+      // "-Xfatal-warnings"
+    ) ++ x.getOrElse(Seq.empty)
+  },
+  excludeDependencies += "org.scala-lang.modules" % "scala-collection-compat_2.13",
+  Compile / console / scalacOptions --= {
+    val x = CrossVersion.partialVersion(scalaVersion.value) flatMap {
+      case (2, _) => Some(Seq("-Xlint"))
+      case _ => None
+    }
+    Seq("-deprecation" /*, "-Xfatal-warnings" */) ++ x.getOrElse(Seq.empty)
+  }
```

maybe just duplicate the options?
```suggestion
  Compile / console / scalacOptions --= CrossVersion.partialVersion(scalaVersion.value) match {
    case Some((2, _)) => Seq("-Xlint", "-deprecation", "-Xfatal-warnings")
    case _ => Seq("-deprecation", "-Xfatal-warnings")
  }
```

---

_when keeping it DRY goes wrong_